app/resources.py

```diff
@@ -0,0 +1,217 @@
+"""
+This module uses Flask-Restful to gives the resources
+access to HTTP methods.
+"""
+from flask import request
+from flask_restful import reqparse, Resource, marshal
+from flask.ext.login import login_required, current_user
+from werkzeug.exceptions import BadRequestKeyError
+
+from sqlalchemy.orm.exc import NoResultFound
+from sqlalchemy_paginator import Paginator
+from sqlalchemy_paginator.exceptions import EmptyPage
+
+from models import BucketList, BucketListItems
+from database import session
+from serializer import bucketlists
+
+
+parser = reqparse.RequestParser()
+
+
+def paging(fields, paginator, page):
+    """
+    Use imported paginator and page arguments to paginate SQLALchemy query
+    sets.The fields argument it takes is used to serialize the results,
+    courtesy of marshal.
+    """
+    try:
+        return marshal(paginator.page(page).object_list, fields), 200
+    except EmptyPage:
+        return {'message': "Page doesn't exist"}, 404
+
+
+def _get_bucketlist(list_id):
+    """Check that the bucketlist belongs to current user."""
+    created_by = current_user.user_id
+    bucketlist = session.query(BucketList).filter_by(
+        list_id=list_id, creator=created_by).first()
+    if not bucketlist:
+        raise NoResultFound
+    return bucketlist
+
+
+def _get_bucketlist_item(item_id, list_id):
+    """Check that a bucketlist item exists."""
+    bucketlistitem = session.query(
+        BucketListItems).filter_by(item_id=item_id, bucket_id=list_id).first()
+    if not bucketlistitem:
+        raise NoResultFound
+    return bucketlistitem
+
+
+class BucketListAll(Resource):
+    """
+    Resource to handle '/bucketlists/' and
+    '/bucketlists/<int:page>' endpoint.
+    """
+
+    @login_required
+    def get(self, page=1):
```

Comment:
**Code Quality: Nitpick**

It's best to implement pagination using GET parameters similar to what you did with the `limit`. Re-implement this bit doing that.